sentinelhub/decoding.py

```diff
@@ -144,12 +145,29 @@ def decode_sentinelhub_err_msg(response: Response) -> str:
     :param response: Sentinel Hub service response
     :return: An error message
     """
+    if not isinstance(response.content, bytes) or not response.content:
+        return ""
+
+    try:
+        json_message = json.loads(response.content)
+        if "error" in json_message:
+            json_message = json_message["error"]
+
+        if isinstance(json_message, str):
+            return json_message
+        return json.dumps(json_message)
+    except JSONDecodeError:
+        pass
+
     try:
         server_message = []
         for elem in decode_data(response.content, MimeType.XML):
-            if "ServiceException" in elem.tag or "Message" in elem.tag:
-                server_message.append(elem.text.strip("\n\t "))
-        return "".join(server_message)
+            if "ServiceException" in elem.tag or "Message" in elem.tag or elem.tag == "body":
+                for text in elem.itertext():
+                    stripped_text = text.strip("\n\t ")
```

Comment:
`strip` already removes all whitespace characters. Is there a reason these are specified by hand?

Reply:
i had to doublecheck as well :smile: